### PR TITLE
Added error handling to useUser [SDK-2236]

### PR DESCRIPTION
**Pull request by Widcket**
### Description

This PR adds an `error` property to the return object of the `useUser` hook, that will contain an `Error` instance when the `fetch()` or `response.json()` calls throw.
The tests, documentation and example apps were updated accordingly.

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [X] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used, if not `master`

### CODE REVIEW

File: tests/fixtures/frontend.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { UserProfile, UserProvider } from '../../src';
+import { UserProvider, UserProviderProps, UserProfile } from '../../src/frontend';
```
Comment (adamjmcgrath) [inline tests/fixtures/frontend.tsx:2]: Could you export `UserProviderProps` from:

https://github.com/auth0/nextjs-auth0/blob/beta/src/index.ts#L95
and
https://github.com/auth0/nextjs-auth0/blob/beta/src/index.browser.ts#L5-L12

Then the developer will be able to do `import { UserProviderProps } from '@auth0/nextjs-auth0';`

And you can leave this import as `'../../src'`

File: src/frontend/with-page-auth-required.tsx [hunk 2]
```diff
@@ -66,18 +81,26 @@ export type WithPageAuthRequired = <P extends object>(
...
+      if ((user && !error) || isLoading) return;
 
       (async (): Promise<void> => {
         await router.push(`${loginUrl}?returnTo=${returnTo}`);
       })();
-    }, [user, isLoading, router, loginUrl, returnTo]);
+    }, [user, error, isLoading]);
+
+    if (error) return onError(error);
+    else if (user) return <Component {...props} />;
```
Comment (adamjmcgrath) [inline src/frontend/with-page-auth-required.tsx:101]: this can be just an `if`

**Review by adamjmcgrath — CHANGES_REQUESTED** [review]

File: tests/fixtures/frontend.tsx (hunk 1 above)
Comment (Widcket) [reply tests/fixtures/frontend.tsx]: Changed in 6ceda646787bd2c0676ecc0a288522b2645e49de.

File: src/frontend/with-page-auth-required.tsx (hunk 2 above)
Comment (Widcket) [reply src/frontend/with-page-auth-required.tsx]: Changed in 6ceda646787bd2c0676ecc0a288522b2645e49de.

**Review by adamjmcgrath — APPROVED** [review]

**Pull request merged**